src/app/file-manager/breadcrumb/breadcrumb.component.ts

```diff
@@ -23,30 +28,43 @@ export class BreadcrumbComponent implements OnInit {
 
   }
 
-  stepClicked(evt, segment) {
-    this.change.emit(segment);
+  stepClicked(evt, segment, index) {
+    this.change.emit(
+      {
+        index: index,
+        item: this.items[index]
+      }
+    );
   }
 
   homeClicked(evt) {
     this.home.emit('home');
   }
 
-  private processPath(fullpath: string) {
-    if (fullpath === null || fullpath === undefined) { return; }
-    fullpath = fullpath.substring(fullpath.length - 1, fullpath.length) === this.DELIMITER
-               ? fullpath.substr(0, fullpath.lastIndexOf(this.DELIMITER)) : fullpath;
-    const segs = fullpath.split(this.DELIMITER);
+  private processItems(steps: Item[]) {
     this.segments = [];
-    let p = '';
-    segs.forEach(s => {
-      p = p + s;
-      const e = {
-        label: s,
-        path: p
-      };
-      if (e.path !== '') {
+
+    if (steps === null || steps === undefined || steps.length <= 0) {
+      return;
+    }
+
+    steps.forEach(
```

Comment:
It looks like the type used here for `steps` doesn't match the parameter declaration `Item[]`. It also looks like this code refers to each element of the breadcrumb as a `step`. I haven't seen that term used before in this context, is this the usual name for this item? I think I've seen code call this element a `crumb` before.